src/main/java/svnserver/ext/gitea/auth/GiteaUserDB.java

```diff
@@ -0,0 +1,161 @@
+/**
+ * This file is part of git-as-svn. It is subject to the license terms
+ * in the LICENSE file found in the top-level directory of this distribution
+ * and at http://www.gnu.org/licenses/gpl-2.0.html. No part of git-as-svn,
+ * including this file, may be copied, modified, propagated, or distributed
+ * except according to the terms contained in the LICENSE file.
+ */
+package svnserver.ext.gitea.auth;
+
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.gitea.ApiClient;
+import io.gitea.ApiException;
+import io.gitea.api.UserApi;
+import io.gitea.auth.Authentication;
+import io.gitea.auth.HttpBasicAuth;
+import io.gitea.model.UserSearchList;
+import svnserver.auth.Authenticator;
+import svnserver.auth.PlainAuthenticator;
+import svnserver.auth.User;
+import svnserver.auth.UserDB;
+import svnserver.context.SharedContext;
+import svnserver.ext.gitea.config.GiteaContext;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Gitea user authentiation.
+ *
+ * @author Artem V. Navrotskiy <bozaro@users.noreply.github.com>
+ * @author Andrew Thornton <zeripath@users.noreply.github.com>
+ */
+public final class GiteaUserDB implements UserDB {
+  @NotNull
+  private static final Logger log = LoggerFactory.getLogger(GiteaUserDB.class);
+  @NotNull
+  private static final String PREFIX_USER = "user-";
+  @NotNull
+  private static final ThreadLocal<ApiClient> threadLocal = new ThreadLocal<>();
+
+  @NotNull
+  private final Collection<Authenticator> authenticators = Collections.singleton(new PlainAuthenticator(this));
+  @NotNull
+  private final GiteaContext context;
+  
+  GiteaUserDB(@NotNull SharedContext context) {
+    this.context = context.sure(GiteaContext.class);
+  }
+
+  @NotNull
+  @Override
+  public Collection<Authenticator> authenticators() {
+    return authenticators;
+  }
+
+  @Override
+  public void updateEnvironment(@NotNull Map<String, String> env, @NotNull User userInfo) {
+    final String externalId = userInfo.getExternalId();
+    env.put("SSH_ORIGINAL_COMMAND", "git");
+    env.put("GITEA_PUSHER_EMAIL", userInfo.getEmail());
+    if (externalId != null) {
+      env.put("GL_ID", PREFIX_USER + externalId);
```

Comment:
Does Gitea really need both GL_ID and GITEA_PUSHER_ID?

Reply:
No. git-lfs-authenticate.cfg mentions it so I left it in. I can get rid of it though.

Reply:
OK I'll kill the GL_* environment variables.